digestparser/json_output.py

```diff
@@ -0,0 +1,185 @@
+"build JSON output from digest content"
+import os
+import time
+import copy
+from collections import OrderedDict
+import requests
+from elifetools.utils import copy_attribute
+from digestparser.utils import formatter_string, msid_from_doi
+from digestparser.jats import (parse_jats_file, parse_jats_digest, parse_jats_pub_date,
+                               parse_jats_subjects, xml_to_html)
+from digestparser.build import build_digest
+
+
+def iiif_server_info(info_url):
+    "get the image info from the IIIF server"
+    info = {}
+    try:
+        response = requests.get(info_url)
+        info = response.json()
+    except:
+        # could be any error right now
+        pass
+    return info
+
+
+def image_info(msid, file_name, digest_config):
+    "get image info from the IIIF server"
+    info_url = formatter_string(digest_config, 'iiif_info_url').format(
+        msid=msid, file_name=file_name)
+    return iiif_server_info(info_url)
+
+
+def image_attribution(credit, image_license):
+    "concatenate an image attribution"
+    attribution = []
+    attribution_line = ', '.join([part for part in [credit, image_license] if part])
+    attribution.append(attribution_line)
+    return attribution
+
+
+def image_uri(msid, file_name, digest_config):
+    "uri of the image file as defined in the settings"
+    return digest_config.get('iiif_image_uri').format(msid=msid, file_name=file_name)
+
+
+def image_source(msid, file_name, digest_config):
+    "source of the iiif image as defined in the settings"
+    source = OrderedDict()
+    source['mediaType'] = "image/jpeg"
+    source['uri'] = digest_config.get('iiif_image_source_uri').format(msid=msid,
+                                                                      file_name=file_name)
+    source['filename'] = file_name
+    return source
+
+
+def image_size(info):
+    "size of the iiif image from the info"
+    size = OrderedDict()
+    # get the size from the json
+    size['width'] = info.get('width')
+    size['height'] = info.get('height')
+    return size
+
+
+def image_json(digest, digest_config):
+    "format image details into JSON format"
+    # need an image file name to continue
+    if not digest.image.file:
```

Comment:
make sense, if you define a `digest` without the image, no image data is added; as soon as you do, failing to build the image part fails _everything_ :+1: